sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuMultiFileBatchReader.java

```diff
@@ -190,6 +190,15 @@ private Schema requiredSchema(GpuDeleteFilter deleteFilter) {
     }
   }
 
+  /**
+   * MultiFiles readers expect the path string is encoded and url safe.
+   * Here leverages this conversion to do this encoding because Iceberg
+   * gives the raw data path by `file().path()` call.
+   */
+  private String toEncodedPathString(FileScanTask fst) {
+    return new Path(fst.file().path().toString()).toUri().toString();
```

Comment:
Does ```new URI``` work here like we did in 
https://github.com/NVIDIA/spark-rapids/pull/9717/files#diff-fddeedd0598bd545d712247c9abf66f4fde2fd0b468a347ff4f2fe9d3c261ac1R322 which is more straightforward?

Reply:
No, here is an exception case.
```
scala> new Path("%2F%23_v9kRtI%27").toUri().toString()
res0: String = %252F%2523_v9kRtI%2527

scala> new URI("%2F%23_v9kRtI%27").toString
res1: String = %2F%23_v9kRtI%27
```